ftp/src/main/scala/akka/stream/alpakka/ftp/impl/FtpBrowserGraphStage.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.ftp
+package impl
+
+import akka.Done
+import akka.stream.stage.{ GraphStageWithMaterializedValue, OutHandler }
+import akka.stream.{ Attributes, Outlet, SourceShape }
+import akka.stream.impl.Stages.DefaultAttributes.IODispatcher
+import scala.concurrent.{ Future, Promise }
+
+private[ftp] trait FtpBrowserGraphStage[FtpClient]
+    extends GraphStageWithMaterializedValue[SourceShape[FtpFile], Future[Done]] {
+
+  def name: String
+
+  def basePath: String
+
+  def connectionSettings: RemoteFileSettings
+
+  implicit def ftpClient: FtpClient
+
+  val ftpLike: FtpLike[FtpClient]
+
+  val shape: SourceShape[FtpFile] = SourceShape(Outlet[FtpFile](s"$name.out"))
+
+  override def initialAttributes: Attributes =
+    super.initialAttributes and Attributes.name(name) and IODispatcher
+
```

Comment:
@johanandren Please, can you confirm that this is what you mean regarding initial attributes?